Submariner doc architecture overhaul stolostron/backlog#27003

oafischer opened the pull request:
https://github.com/stolostron/backlog/issues/27003

oafischer committed:
Submariner doc architecture overhaul

oafischer committed:
Update main subm architecture

oafischer committed:
Fix link

oafischer committed:
Fix links

oafischer committed:
Fix link

oafischer committed:
Fix link

swopebe commented:
I will return to this after the break...amazing work!

sridhargaddam commented:
Nice work @oafischer. I will take a look at it during this week.

oafischer commented:
Awesome, thanks @sridhargaddam! I will walk through some of the architecture stuff with @dfarrell07 as well but this is a great opportunity to review the whole docs for technical accuracy and such. I made some small content updates but I'm sure there's still stuff that is outdated or that could be improved, if you want to focus on that.

swopebe commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -59,4 +84,4 @@ It might take several minutes for the installation and configuration to complete
 * `Gateway nodes labeled` indicates how many worker nodes are labeled with the Submariner gateway label: `submariner.io/gateway=true` on the managed cluster.
 
-Submariner is now deployed on the clusters.
+Submariner is now deployed on the seclected clusters.
```
spelling here line 87

swopebe reviewed (approved):
One small comment and I also think the #5 per style should be spelled out (I haven't looked at that in a while), but other than that, this is huge iterative improvements on this doc.

sridhargaddam commented on add-ons/addon_overview.adoc:
```diff
@@ -3,7 +3,7 @@
 
 {product-title} add-ons can improve some areas of performance and add function to enhace your applications. The following sections provide a summary of the add-ons that are available for {product-title-short}:
```
```suggestion
{product-title} add-ons can improve some areas of performance and add functionality to enhance your applications. The following sections provide a summary of the add-ons that are available for {product-title-short}:
```

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -1,19 +1,27 @@
 *Note:* Only non-NSX deployments are supported on VMware vSphere.
 
-To deploy Submariner on other providers, follow the instructions in xref:../submariner/submariner_deploy_manual.adoc#deploying-submariner-manually[Deploying Submariner manually]. Complete the following steps to deploy Submariner with the {product-title} console:
+To deploy Submariner on other providers, follow the instructions in xref:../submariner/deploy_subm_manual.adoc#preparing-submariner-manual-deployment[Deploying Submariner hosts manually].
```
```suggestion
To deploy Submariner on other providers, follow the instructions in xref:../submariner/deploy_subm_manual.adoc#preparing-submariner-manual-deployment[Deploying Submariner manually].
```

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -1,19 +1,27 @@
 = Deploying Submariner with the console
 
-You can deploy Submariner on {ocp} managed clusters that are deployed on Amazon Web Services, Google Cloud Platform, and VMware vSphere by using the {product-title} console.
+Before you deploy Submariner with {product-title}, you must prepare the clusters on the hosting environment for the connection. You can use the `SubmarinerConfig` API to automatically prepare clusters to deploy Submariner on {ocp} managed clusters that are deployed on the following providers:
```
```suggestion
Before you deploy Submariner with {product-title}, you must prepare the clusters on the hosting environment. You can use the `SubmarinerConfig` API or the {product-title} console to automatically prepare {ocp} clusters on the following providers:
```

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -35,15 +43,32 @@ To deploy Submariner on other providers, follow the instructions in xref:../subm
 . Select the clusters on which you want to deploy Submariner. 
 
-. Enter the following information in the _Install Submariner add-ons_ editor:
+. See the fields in following table and enter the required information in the _Install Submariner add-ons_ editor:
```
```suggestion
. See the fields in the following table and enter the required information in the _Install Submariner add-ons_ editor:
```

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -35,15 +43,32 @@ To deploy Submariner on other providers, follow the instructions in xref:../subm
+| The default value for the IPsec NAT traversal port is port `4500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
+
+| `Gateway count`
+| The number of worker nodes that are used to deploy the Submariner gateway component on your managed cluster. The default value is `1`. If the value is greater than 1, the Submariner gateway High Availability (HA) is automatically enabled.
```
JFYI, for the following cluster (AWS, GCP, Azure and OpenStack) we always create new dedicated Gateway nodes. Only for VmWare clusters we label one (or more) of the existing worker nodes as Gateway nodes.

```suggestion
| The number of gateway nodes to be deployed on the managed cluster. For AWS, GCP, Azure and OpenStack clusters dedicated Gateway nodes are deployed whereas for VmWare clusters, existing worker nodes are tagged as Gateway nodes. The default value is `1`. If the value is greater than 1, the Submariner gateway High Availability (HA) is automatically enabled.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -59,4 +84,4 @@ It might take several minutes for the installation and configuration to complete
 +
 * `Gateway nodes labeled` indicates how many worker nodes are labeled with the Submariner gateway label: `submariner.io/gateway=true` on the managed cluster.
```
```suggestion
* `Gateway nodes labeled` indicates the number of gateway nodes on the managed cluster.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+[#preparing-submariner-manual-deployment]
+= Deploying Submariner hosts manually
+
+Before you deploy Submariner with {product-title}, you must prepare the clusters on the hosting environment for the connection. See xref:../submariner/submariner_prepare_hosts.adoc#preparing-selected-hosts-to-deploy-submariner[Deploying Submariner with the console] to learn how to automatically deploy Submariner on supported clusters by using the console.
```
```suggestion
Before you deploy Submariner with {product-title}, you must prepare the clusters on the hosting environment for the connection. See xref:../submariner/submariner_prepare_hosts.adoc#preparing-selected-hosts-to-deploy-submariner[Deploying Submariner using? the console] to learn how to automatically deploy Submariner on supported clusters by using the console.
```

swopebe replied:
by using (otherwise it looks like submariner is using the console)

oafischer replied:
Can we say "with the console" since we seem to use that in our docs? Or should we change to "by using"?

swopebe replied:
Well, I think it can be read as:

Deploying Submariner with the console vs Deploying Submariner without the console... 

But we are using the console to deploy submariner, which is prob the reason for the change request. 

Per IBM Style, we should be writing "by using" for that clarity, so you may not find that consistency in the doc...but it is the goal.

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+Before you deploy Submariner with {product-title}, you must prepare the clusters on the hosting environment for the connection. See xref:../submariner/submariner_prepare_hosts.adoc#preparing-selected-hosts-to-deploy-submariner[Deploying Submariner with the console] to learn how to automatically deploy Submariner on supported clusters by using the console.
+
+If your cluster does not support automatic deployment, see the following topics to prepare your selected hosting environment manually for deployment on Submariner. Each host has unique steps for preparation, so make sure to select the correct host.
```
```suggestion
If your cluster is hosted on a provider that does not support automatic Submariner deployment, see the following sections to prepare the infrastructure manually. Each provider has unique steps for preparation, so make sure to select the correct one.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+There are two kinds of Red Hat OpenShift Kubernetes Service (ROKS) on IBM Cloud: the classic cluster and the second generation of compute infrastructure in a virtual private cloud (VPC). Submariner cannot run on the classic ROKS cluster since it cannot configure the IPsec ports for the classic cluster.
+
+To configure the ROKS clusters on a VPC to use Submariner, complete the steps in the following links:
```
```suggestion
To configure the ROKS clusters on a VPC to use Submariner, complete the following steps:
```

sridhargaddam reviewed (changes requested):
Thanks @oafischer for the PR.

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+To configure the ROKS clusters on a VPC to use Submariner, complete the steps in the following links:
+
+. Before you create a cluster, specify subnets for pods and services, which avoids overlapping CIDRs with other clusters. Make sure there are no overlapping pods and services CIDRs between clusters if you are using an existing cluster. See https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-subnets#vpc_basics[VPC Subnets] for the procedure.
```
```suggestion
. If you plan to install Submariner without Globalnet, make sure that pod and service CIDRs do not overlap between clusters in the clusterSet. See https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-subnets#vpc_basics[VPC Subnets] for the procedure. For clusters with overlapping CIDRs, Globalnet can be enabled while installing Submariner.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+. Attach a public gateway to subnets used in the cluster. See https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-subnets#vpc_basics_pgw[Public Gateway] for the procedure.
+
+. Create inbound rules for the default security group of the cluster by completing the steps in https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-network-policy#security_groups_ui[Security Group]. Ensure that the firewall allows inbound and outbound traffic on 4500/UDP and 500/UDP ports for Gateway nodes, and allows inbound and outbound UDP/4800 for all the other nodes.
```
```suggestion
. Create inbound rules for the default security group of the cluster by completing the steps in https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-network-policy#security_groups_ui[Security Group]. Ensure that the firewall allows inbound and outbound traffic for external clients on 4500/UDP, 4490/UDP ports for the Gateway nodes, and allows inbound and outbound UDP/4800 traffic within the local cluster between the nodes.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+. Create inbound rules for the default security group of the cluster by completing the steps in https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-network-policy#security_groups_ui[Security Group]. Ensure that the firewall allows inbound and outbound traffic on 4500/UDP and 500/UDP ports for Gateway nodes, and allows inbound and outbound UDP/4800 for all the other nodes.
+
+. Label a node that has the public gateway as `submariner.io/gateway=true` in the cluster.
```
```suggestion
. Label the node selected as Gateway node in the previous step with the following label `submariner.io/gateway=true`.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+To prepare bare metal clusters for deploying Submariner, complete the following steps:
+
+. Ensure that the IPsec NATT and VXLAN ports are open.
```
```suggestion
. Ensure that inbound and oubound VxLAN (UDP/4800) traffic between the cluster nodes is allowed in the firewall configuration.
```

sridhargaddam replied:
At this step, we do not know the Gateway node, so lets add it as an extra step below.

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+Submariner uses IP security (IPsec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPsec NATT port, or you can specify a different port that you configured. When you run this procedure without specifying an IPsec NATT port of 4500/UDP is automatically used for the communication.
+
+See xref:../submariner/submariner_customizations.adoc#submariner-customizations[Customizing Submariner deployments] for information about the customization options. 
+
```
Please add this step 

. Identify the Gateway node configured by Submariner and enable firewall configuration to allow the IPsec NATT (UDP/4500), NatDiscovery (UDP/4490) ports for external traffic.

oafischer replied:
Added

oafischer committed:
Update discovery_config_ui.adoc

oafischer committed:
Update add-ons/addon_overview.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_console.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_console.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_console.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_console.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_console.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_console.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_console.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_manual.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_manual.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer commented:
Thanks for your review @sridhargaddam! I agreed with your changes and committed them with edits where necessary. Since this is a high priority PR, can you take one more look so that I can merge it soon? Some other Submariner work depends on these changes. Thanks again!

oafischer committed:
Update uninstall_subm.adoc

oafischer committed:
Update deploy_subm_manual.adoc

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -0,0 +1,87 @@
+
+| `Google Cloud Platform service account JSON key`
+| Only visible when you import a Google Cloud Platform cluster.
+
```
@aswinsuryan @vthapar do you remember the fields for the Azure clusters?

MaxBab replied:
@sridhargaddam 

Azure clusters has the following fields:
`Base domain resource group name`
`Client ID`
`Client secret`
`Subscription ID`
`Tenant ID`

oafischer replied:
Added the extra fields, table is kind of long, hope that's OK. Maybe in the future if it gets too long we can find a way to restructure.

sridhargaddam replied:
thanks @MaxBab

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -0,0 +1,87 @@
+| Only visible when you import a Google Cloud Platform cluster.
+
+| `Instance type`
+| The Amazon Web Services EC2 instance type of the gateway node that is created on the managed cluster. The default value is `c5d.large`. This field is only visible when your managed cluster environment is AWS.
```
The instance type is applicable for other clusters as well and not just AWS. If we want to capture the default value, IMO we have to do the same for all the other clusters. So, I personally feel that we can avoid capturing the default value here.

```suggestion
| The instance type of the gateway node that is created on the managed cluster.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -0,0 +1,87 @@
+| The Amazon Web Services EC2 instance type of the gateway node that is created on the managed cluster. The default value is `c5d.large`. This field is only visible when your managed cluster environment is AWS.
+
+| `IPsec NAT-T port`
+| The default value for the IPsec NAT traversal port is port `4500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewalls.
```
```suggestion
| The default value for the IPsec NAT traversal port is port `4500`. If your managed cluster environment is VMware vSphere, ensure that this port is opened on your firewall.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+[#preparing-submariner-manual-deployment]
+= Deploying Submariner hosts manually
```
```suggestion
= Deploying Submariner manually
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+To prepare bare metal clusters for deploying Submariner, complete the following steps:
+
+. Ensure that inbound and oubound VxLAN (UDP/4800) traffic between the cluster nodes is allowed in the firewall configuration.
```
```suggestion
. Ensure that the firewall allows inbound/outbound traffic for external clients on 4500/UDP, 4490/UDP ports for the Gateway nodes, and allows inbound/outbound UDP/4800 traffic within the local cluster nodes.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+
+. Attach a public gateway to subnets used in the cluster. See https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-subnets#vpc_basics_pgw[Public Gateway] for the procedure.
+
+. Create inbound rules for the default security group of the cluster by completing the steps in https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-network-policy#security_groups_ui[Security Group]. Ensure that the firewall allows inbound and outbound traffic for external clients on 4500/UDP, 4490/UDP ports for the Gateway nodes, and allows inbound and outbound UDP/4800 traffic within the lo ...
```
```suggestion
. Create inbound rules for the default security group of the cluster by completing the steps in https://cloud.ibm.com/docs/openshift?topic=openshift-vpc-network-policy#security_groups_ui[Security Group]. Ensure that the firewall allows inbound/outbound traffic for external clients on 4500/UDP, 4490/UDP ports for the Gateway nodes, and allows inbound/outbound UDP/4800 traffic within the local cluster nodes.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+spec:{}
+----
++
+Replace `managed-cluster-namespace` with the namespace of your managed cluster. The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
```
```suggestion
Replace `managed-cluster-namespace` with the name of your managed cluster. The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
++
+Replace `managed-cluster-namespace` with the namespace of your managed cluster. The name of the `SubmarinerConfig` must be `submariner`, as shown in the example.
++
+This configuration uses the default network address translation - traversal (NATT) port (4500/UDP) for your Submariner and one worker node is labeled as the Submariner gateway on your bare metal cluster.
```
```suggestion
This configuration labels one of the worker node as the Submariner gateway on your bare metal cluster.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
++
+This configuration uses the default network address translation - traversal (NATT) port (4500/UDP) for your Submariner and one worker node is labeled as the Submariner gateway on your bare metal cluster.
++
+Submariner uses IP security (IPsec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPsec NATT port, or you can specify a different port that you configured. When you run this procedure without specifying an IPsec NATT port of 4500/UDP is automatically used for the communication.
```
```suggestion
By default, Submariner uses IP security (IPsec) to establish the secure tunnels between the clusters on the gateway nodes. You can either use the default IPsec NATT port, or you can specify a different port that you configured. When you run this procedure without specifying an IPsec NATT port, 4500/UDP is used for the connections.
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+[#deploying-submariner-mcaddon-api]
+== Deploy Submariner with the ManagedClusterAddOn API
+
+After manually preparing your selected hosting environment, you can deploy Submariner with the `ManagedClusterAddOn` API by complete the following steps:
```
```suggestion
After manually preparing your selected hosting environment, you can deploy Submariner with the `ManagedClusterAddOn` API by completing the following steps:
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
++
+Replace `managed-cluster-set-name` with a name for the `ManagedClusterSet` that you are creating.
++
+*Note:* The maximum length of the name of the Kubernetes namespace is 63 characters, so the maximum length of the `<managed-cluster-set-name>` is 56 characters. If the length of `<managed-cluster-set-name>` exceeds 56, the `<managed-cluster-set-name>` is cut off from the head.
```
```suggestion
*Note:* The maximum length of a Kubernetes namespace can be 63 characters. It is recommended that the maximum length of the `<managed-cluster-set-name>` is 56 characters or less. If the length of `<managed-cluster-set-name>` exceeds 56 chars, the `<managed-cluster-set-name>` is cut off from the head.
```

oafischer replied:
As per IBM style, "can" indicates an ability. I'd consider the character limit more of a limit than an ability. I also wouldn't say that 56 characters or fewer is a recommendation if it's also a hard limitation. I'll make a couple changes to the wording but I think we need to keep most of it as is.

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+     labels:
+         cluster.open-cluster-management.io/backup: submariner
+spec:
+     globalnetEnabled: false
```
```suggestion
     globalnetEnabled: <true-or-false>
```

sridhargaddam commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,253 @@
+spec:{}
+----
++
+Replace `managed-cluster-namespace` with the namespace of your managed cluster.
```
A User may not be aware that we create a namespace for every managed cluster. Since a namespace is created with the same name as the managed cluster, shall we rephrase the sentence as below?

CC: @nyechiel @aswinsuryan @vthapar 

```suggestion
Replace `managed-cluster-namespace` with the name of your managed cluster.
```

sridhargaddam reviewed (changes requested):
Thanks for addressing the previous comments @oafischer

sridhargaddam commented on add-ons/submariner/subm_configure.adoc:
```diff
@@ -0,0 +1,91 @@
+* A credential to access the hub cluster with `cluster-admin` permissions.
+* IP connectivity must be configured between the Gateway nodes. When connecting two clusters, at least one of the clusters must be accessible to the Gateway node using its public or private IP address designated to the Gateway node. See https://submariner.io/operations/nat-traversal[Submariner NAT Traversal] for more information.
+* If you are using OVN Kubernetes, clusters must be at {ocp} version 4.11 or later. 
+* Firewall configuration across all nodes in each of the managed clusters must allow 4800/UDP in both directions.
```
```suggestion
* If your OCP clusters use OpenShift SDN CNI, the firewall configuration across all nodes in each of the managed clusters must allow 4800/UDP in both directions.
```

sridhargaddam commented on add-ons/submariner/subm_configure.adoc:
```diff
@@ -0,0 +1,91 @@
+* IP connectivity must be configured between the Gateway nodes. When connecting two clusters, at least one of the clusters must be accessible to the Gateway node using its public or private IP address designated to the Gateway node. See https://submariner.io/operations/nat-traversal[Submariner NAT Traversal] for more information.
+* If you are using OVN Kubernetes, clusters must be at {ocp} version 4.11 or later. 
+* Firewall configuration across all nodes in each of the managed clusters must allow 4800/UDP in both directions.
+* Firewall configuration on the Gateway nodes must allow ingress 8080/TCP so that the other nodes in the cluster can access it.
```
This step is no longer needed.

```suggestion
```

sridhargaddam commented on add-ons/submariner/subm_configure.adoc:
```diff
@@ -0,0 +1,91 @@
+* If you are using OVN Kubernetes, clusters must be at {ocp} version 4.11 or later. 
+* Firewall configuration across all nodes in each of the managed clusters must allow 4800/UDP in both directions.
+* Firewall configuration on the Gateway nodes must allow ingress 8080/TCP so that the other nodes in the cluster can access it.
+* Firewall configuration must be open for 4500/UDP and any other ports that are used for IPsec traffic on the gateway nodes. 
```
```suggestion
* Firewall configuration must allow 4500/UDP, 4490/UDP on the gateway nodes for establishing tunnels between the managed clusters. 
```

sridhargaddam commented on add-ons/submariner/subm_configure.adoc:
```diff
@@ -0,0 +1,91 @@
+* Firewall configuration on the Gateway nodes must allow ingress 8080/TCP so that the other nodes in the cluster can access it.
+* Firewall configuration must be open for 4500/UDP and any other ports that are used for IPsec traffic on the gateway nodes. 
++
+*Note:* This is configured automatically when your clusters are deployed in an AWS or GCP environment, but must be configured manually for clusters on other environments and for the firewalls that protect private clouds.
```
```suggestion
*Note:* This is configured automatically when your clusters are deployed in an AWS, GCP, Azure and Red Hat OpenStack environments, but must be configured manually for clusters on other environments and for the firewalls that protect private clouds.
```

sridhargaddam commented on add-ons/submariner/subm_configure.adoc:
```diff
@@ -0,0 +1,91 @@
+| 4500/UDP
+| Yes
+| Required
+
```
Please add one more entry as shown below.

| NAT discovery port
| 4490/UDP
| No
| Required

sridhargaddam commented on add-ons/submariner/subm_configure.adoc:
```diff
@@ -0,0 +1,91 @@
+
+Globalnet is a feature included with the Submariner add-on which supports connectivity between clusters with overlapping CIDRs. Globalnet is a cluster set wide configuration, and can be selected when the first managed cluster is added to the cluster set. When Globalnet is enabled, each managed cluster is allocated a global CIDR from the virtual Global Private Network. The global CIDR is used for  ...
+
+If there is a chance that your clusters running Submariner might have overlapping CIDRs, consider enabling Globalnet. When using the console, the `ClusterAdmin` can enable Globalnet for a cluster set by selecting the option *Enable Globalnet* when enabling the Submariner add-on for clusters in the cluster set. After you enable Globalnet, you cannot disable it without removing Submariner.
```
```suggestion
If there is a chance that your clusters running Submariner might have overlapping CIDRs, consider enabling Globalnet. When using the console, the `ClusterAdmin` can enable Globalnet for a clusterSet by selecting the option *Enable Globalnet* when enabling the Submariner add-on for clusters in the clusterSet. After you enable Globalnet, you cannot disable it without removing Submariner.
```

oafischer replied:
We refer to it as cluster set in the docs, unless we are talking about a specific value or parameter.

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_manual.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_console.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_console.adoc

oafischer committed:
Update subm_intro.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_console.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_manual.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_manual.adoc

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/deploy_subm_manual.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update deploy_subm_manual.adoc

oafischer committed:
Update add-ons/submariner/subm_configure.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update subm_configure.adoc

oafischer committed:
Update add-ons/submariner/subm_configure.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update add-ons/submariner/subm_configure.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update subm_configure.adoc

oafischer committed:
Update add-ons/submariner/subm_configure.adoc

Co-authored-by: Sridhar Gaddam <sgaddam@redhat.com>

oafischer committed:
Update subm_configure.adoc

oafischer committed:
Update install_subctl.adoc

oafischer committed:
Update uninstall_subm.adoc

oafischer committed:
Update uninstall_subm.adoc

oafischer committed:
Update subm_intro.adoc

oafischer committed:
Update service_discovery.adoc

oafischer committed:
Update deploy_subm_manual.adoc

oafischer committed:
Update subm_configure.adoc

oafischer committed:
Update subm_intro.adoc

oafischer committed:
Update subm_intro.adoc

oafischer committed:
Update subm_intro.adoc

oafischer commented:
@sridhargaddam Thanks for another review! I implemented most of your suggestions and fixed a number of broken links as well. Hoping we can merge this soon if you or @dfarrell07 or @nyechiel want to take a last look.

MaxBab commented on add-ons/submariner/deploy_subm_console.adoc:
```diff
@@ -0,0 +1,87 @@
+
+| `Cable driver`
+| The Submariner gateway cable engine component that maintains the cross-cluster tunnels. The default value is `Libreswan IPsec`.
+|===
```
In additional to the above fields, the following new fields were introduced in ACM 2.7 release for every platform:

| `Disconnected cluster`
| Enable this option to instruct Submariner not to access any external servers for public IP resolution.

| `Globalnet CIDR`
| The Globalnet CIDR to be used for the managed cluster (If left blank, a CIDR will be allocated from clusterset pool).

*Note* - The `Globalnet CIDR` parameter appears only when *Globalnet* configuration selected on the clusterset.

MaxBab commented on add-ons/submariner/deploy_subm_manual.adoc:
```diff
@@ -0,0 +1,255 @@
+See xref:../submariner/submariner_customizations.adoc#customizing-submariner[Customizing Submariner deployments] for information about the customization options. 
+
+[#preparing-aro-subariner]
+== Preparing Microsoft Azure Red Hat OpenShift for Submariner (Technology Preview)
```
As far as I know ARO is not a tech preview and going to be fully supported in the upcoming release.
So need to remove the tech preview note.

@nyechiel ^^

oafischer replied:
If it is fully supported we will need to update the support matrix as well, and one or two sections in this doc.

sridhargaddam replied:
I'll let @nyechiel comment about it, but AFAIR both ARO/ROSA (which use SM LB mode) are tech preview.

MaxBab commented on add-ons/submariner/install_subctl.adoc:
```diff
@@ -0,0 +1,42 @@
+. Download the https://catalog.redhat.com/software/containers/rhacm2/subctl-rhel8/6229131e49e7196373df7d3e[`subctl` container] and extract a compressed version of the `subctl` binary to `/tmp` by entering the following command:
++
+----
+oc image extract registry.redhat.io/rhacm2/subctl-rhel8:v0.13 --path=/dist/subctl-v0.13.0-linux-amd64.tar.xz:/tmp/ --confirm
```
Need to update the version of the subctl from v0.13 to v0.14 here and in the following lines below:
12, 17, 23

oafischer replied:
Done

oafischer committed:
Update deploy_subm_console.adoc

oafischer committed:
Update deploy_subm_console.adoc

oafischer committed:
Update deploy_subm_console.adoc

oafischer committed:
Update install_subctl.adoc

oafischer commented:
Thanks for approving @sridhargaddam, I'm going to merge so I can work on other Submariner issues. If we still need to update some sections (like tech preview) we can fix them later (probably in a new issue).
